lib/rucio/richclient.py

```diff
@@ -0,0 +1,283 @@
+            logger_level = logging.DEBUG
+
+        # overriding by the config
+        cfg_levels = (logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR)
+        for level in cfg_levels:
+            cfg_opt = 'forceloglevel' + logging.getLevelName(level)
+            if _force_cfg_log_level(cfg_opt):
+                logger_level = level
+
+    # setting the log level
+    logger.setLevel(logger_level)
+
+    # preferred logger handling
+    def add_handler(logger: logging.Logger,
+                    console: Optional[Console] = None,
+                    verbose: bool = False) -> None:
+
+        def time_formatter(timestamp: datetime) -> Text:
+            formatted_time = timestamp.strftime("%Y-%m-%d %H:%M:%S")
+            milliseconds = timestamp.microsecond // 1000
+            return Text(f"[{formatted_time},{milliseconds:03d}]")
+
+        console = console or Console()
+        highlighter = ReprHighlighter()
+        handler = RichHandler(rich_tracebacks=True, markup=True, show_path=verbose, show_time=verbose, console=console, tracebacks_width=min(console.width, MAX_TRACEBACK_WIDTH),
+                              tracebacks_word_wrap=True, log_time_format=time_formatter, highlighter=highlighter)
+        logger.addHandler(handler)
+
+    # setting handler and formatter
+    if not logger.handlers:
+        console = console or Console()
+        add_handler(logger, console, verbose)
+
+    return logger
+
+
+def generate_table(table_format: box.Box, headers: list[str], rows: Sequence[Union[tuple[Any, Any], list[Any]]],
+                   col_alignments: Optional[list[JustifyMethod]] = None, show_header: bool = True,
+                   row_styles: Optional[list[str]] = None) -> Table:
+    """
+    Generates a rich Table object from given input rows.
+    :param table_format: style of the table
+    :param headers: list of headers
+    :param rows: list of rows
+    :param col_alignments: list of column alignments
+    :param show_header: boolean to show header
+    :param row_styles: list of row styles
+    :returns: rich Table object
+    """
+    table = Table(box=table_format, show_header=show_header)
+    rich_widgets = (Text, Tree, Table)
+    if row_styles is None:
+        row_styles = ['none', 'dim']
+    table.row_styles = row_styles
+
+    if not col_alignments and len(rows) > 0:
+        # auto-detect on first row, numerical values on the right
+        col_alignments = ['right' if str(value).isnumeric() else 'left' for value in rows[0]]
+    elif col_alignments and len(rows) > len(col_alignments):
+        while len(rows) > len(col_alignments):
+            col_alignments.append('left')
+
+    for i, header in enumerate(headers):
+        if col_alignments:
+            justify = col_alignments[i] if col_alignments[i] in ["left", "center", "right"] else "left"
+            table.add_column(header, overflow='fold', justify=justify)
+        else:
+            table.add_column(header, overflow='fold')
```

Comment:
1. Both single and double quotes. Please pick one and apply it consistently.
 2. No need for the if-else structure. Calling `add_column()` has `justify='left'` as a default.

Reply:
I believe this still stands? `col_alignments` is, due to the previous block, filed with `['left', 'left', …]` unless set as an arguments. What prevents you from calling `table.add_column(header, overflow='fold', justify=justify)` unconditionally?

Reply:
My initial thought process was that if the user accidentally provides a list `col_alignments` that is shorter than the elements in a row, the method could still fill the "missing" alignments with the default value `'left'` instead of breaking. E.g. you could have `row=['filled', 'test', 'row']` and `col_alignments=['left', 'right']` leading to `col_alignments=['left', 'right', 'left']`.

But yes I realised that you are right that if `col_alignments` will always be true since I always set it (if not provided) or fill the missing values. I will fix this.